### PR TITLE
Keyboard Navigation Support for `base-map` Layer Selector

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
The layer selector was previously not accessible by keyboard. This PR rectifies this.

### CODE REVIEW

File: packages/base-map/src/index.tsx [hunk 1]
```diff
@@ -151,6 +151,8 @@ const BaseMap = ({
         <Styled.LayerSelector
           className="filter-group"
           id="filter-group"
+          tabIndex={0}
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:154]: Sort props (multiple instances)

**Comment by binh-dam-ibigroup** [comment]
In the Optional Layers story, when the keyboard focus moves from the layer list to outside of it, the list remains displayed, and there is no way to dismiss it.

**Comment by miles-grant-ibigroup** [comment]
Is that a problem?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I think what needs to happen with the layer selector to support broader accessibility is as follows: add a `<button>` (with labels, aria attributes etc) to open/close the list.  Or, try with a `<details>`/`<summary>` element to have the expanding features built in.

**Comment by miles-grant-ibigroup** [comment]
I think we might want to have an offline design call about this -- I see this as a list that's always present, and only "hidden" via hover when a mouse/touch is present. I don't think it should ever be hidden behind a button.

**Comment by binh-dam-ibigroup** [comment]
> Is that a problem?

Yes, if I'm zoomed in (or have a small screen) then I can't see much of the map anymore.

**Comment by miles-grant-ibigroup** [comment]
Even at high zoom levels and small screen size, I'm still able to clearly see the map. Screenshot 2023-02-17 at 12 32 22 PM

The bigger issue in my eyes is the fake keyboard target at the top of the map. I've been trying to remove this, but can't find a way to do it without adding many extra elements... Will keep trying

**Comment by miles-grant-ibigroup** [comment]
@binh-dam-ibigroup I've supported layers hiding via keyboard, but it's not pretty. Let me know what you think.

File: packages/base-map/src/index.tsx [hunk 2]
```diff
@@ -151,9 +151,29 @@ const BaseMap = ({
         <Styled.LayerSelector
           className="filter-group"
           id="filter-group"
+          onFocus={() => setFakeHover(true)}
+          onBlur={e => {
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:155]: There is still something going on with the blue handling. Using the story at http://localhost:5555/?path=/story/basemap--optional-layers&globals=locale:en-US, you cannot tab to toggle the second layer.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
See https://github.com/opentripplanner/otp-ui/pull/546/files#r1115003171 which is blocking.

File: packages/base-map/src/index.tsx (hunk 2 above)
Comment (binh-dam-ibigroup) [reply packages/base-map/src/index.tsx:155]: On the story at http://localhost:5555/?path=/story/basemap--with-multiple-base-layers, the popup still cannot be dismissed.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
See https://github.com/ibi-group/otp-ui/pull/12.

**Comment by miles-grant-ibigroup** [comment]
Thanks again! Much cleaner

**Review by binh-dam-ibigroup — APPROVED** [review]

File: packages/base-map/src/index.tsx [hunk 3]
```diff
@@ -151,9 +151,9 @@ const BaseMap = ({
         <Styled.LayerSelector
           className="filter-group"
           id="filter-group"
```
Comment (amy-corson-ibigroup) [inline packages/base-map/src/index.tsx:153]: Any chance we could add a `tabIndex` here to take care of the focus order within the map? I think the `LayerSelector` should receive focus before the map attributions if possible!!

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good to me! Just a nitpick about the focus order:

File: packages/base-map/src/index.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/base-map/src/index.tsx:153]: I think the difficulty here is how we get the focus into the selector to begin with. I'll make an issue from this for anyone in the future who'd like to improve this!

**Pull request merged**